flow360/component/volume_mesh.py

```diff
@@ -772,3 +786,192 @@ def __getitem__(self, index) -> VolumeMesh:
+            if not mesh_parser.is_valid_volume_mesh():
+                raise Flow360ValueError(
+                    f"Unsupported volume mesh file extensions: {mesh_parser.format.ext()}. "
+                    f"Supported: [{MeshFileFormat.UGRID.ext()},{MeshFileFormat.CGNS.ext()}]."
+                )
+            self.format = mesh_parser.format
+
+        if self.project_name is None:
+            self.project_name = os.path.splitext(os.path.basename(self.file_name))[0]
+            log.warning(
+                "`project_name` is not provided. "
+                f"Using the volume mesh file name {self.project_name} as project name."
+            )
+
+        if self.length_unit not in LengthUnitType.__args__:
+            raise Flow360ValueError(
+                f"specified length_unit : {self.length_unit} is invalid. "
+                f"Valid options are: {list(LengthUnitType.__args__)}"
+            )
+
+        if self.solver_version is None:
+            raise Flow360ValueError("solver_version field is required.")
+
+    # pylint: disable=protected-access
+    def submit(self, description="", progress_callback=None) -> VolumeMeshV2:
+        """
+        Submit volume mesh to cloud and create a new project
+
+        Parameters
+        ----------
+        description : str, optional
+            description of the project, by default ""
+        progress_callback : callback, optional
+            Use for custom progress bar, by default None
+
+        Returns
+        -------
+        VolumeMeshV2
+            Volume mesh object with id
+        """
+
+        self._validate()
+
+        if not shared_account_confirm_proceed():
+            raise Flow360ValueError("User aborted resource submit.")
+
+        req = NewVolumeMeshRequestV2(
+            name=self.project_name,
+            solver_version=self.solver_version,
+            tags=self.tags,
+            file_name=os.path.basename(self.file_name),
+            # pylint: disable=fixme
+            # TODO: remove hardcoding
+            parent_folder_id="ROOT.FLOW360",
+            length_unit=self.length_unit,
+            format=self.format.value,
+            description=description,
+        )
+
+        ##:: Create new Volume mesh resource and project
+        resp = RestApi(VolumeMeshInterfaceV2.endpoint).post(req.dict())
+        info = VolumeMeshMetaV2(**resp)
+
+        ##:: upload volume mesh file
+        volume_mesh = VolumeMeshV2(info.id)
+        heartbeat_info = {"resourceId": info.id, "resourceType": "VolumeMesh", "stop": False}
+        # Keep posting the heartbeat to keep server patient about uploading.
+        heartbeat_thread = threading.Thread(target=post_upload_heartbeat, args=(heartbeat_info,))
+        heartbeat_thread.start()
+        volume_mesh._webapi._upload_file(
+            remote_file_name=os.path.basename(self.file_name),
+            file_name=self.file_name,
+            progress_callback=progress_callback,
+        )
```

Comment:
how about compress and upload? I think we want to keep this option if file is not compressed before:
https://github.com/flexcompute/Flow360/blob/f6599cf7d1f50240f48d24e7ffd90c6da9cf8ab0/flow360/component/volume_mesh.py#L423

Reply:
Fixed in [7f0098b](https://github.com/flexcompute/Flow360/pull/470/commits/7f0098ba93697f090bbf2efe881718efd3fde64a)